docs/source/tutorials/knn_outlier.ipynb

```diff
@@ -0,0 +1,447 @@
+    "\n",
+    "from keras.datasets import cifar10, cifar100\n",
+    "from sklearn.model_selection import train_test_split"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1ae35d12",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load cifar10 datasets\n",
+    "(trainX, trainy), (testX, testy) = cifar10.load_data()\n",
+    "trainX, _, trainy, _ = train_test_split(trainX, trainy, test_size=0.5, stratify=trainy, random_state=SEED) # only use 50% of each label\n",
+    "\n",
+    "# Convert from integers to floats and normalize range 0-1\n",
+    "trainX = trainX.astype('float32') / 255.0\n",
+    "testX = testX.astype('float32') / 255.0\n",
+    "\n",
+    "# Manually remove non-animals out of the training dataset\n",
+    "animal_labels = [2,3,4,5,6,7]\n",
+    "animal_idxs = np.where(np.isin(trainy, animal_labels))[0] # find idx of animals\n",
+    "trainX = trainX[animal_idxs]\n",
+    "trainy = trainy[animal_idxs]\n",
+    "\n",
+    "# Check the shapes of our training and test sets\n",
+    "print('Train: X=%s, y=%s' % (trainX.shape, trainy.shape))\n",
+    "print('Test: X=%s, y=%s' % (testX.shape, testy.shape))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4480c85",
+   "metadata": {},
+   "source": [
+    "#### Lets visualize some of the training and test examples"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94a7bb8a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "txt_labels = {0: 'airplane', \n",
+    "              1: 'automobile', \n",
+    "              2: 'bird',\n",
+    "              3: 'cat', \n",
+    "              4: 'deer', \n",
+    "              5: 'dog', \n",
+    "              6: 'frog', \n",
+    "              7: 'horse', \n",
+    "              8:'ship', \n",
+    "              9:'truck'}\n",
+    "\n",
+    "def plot_images(X,y):\n",
+    "    plt.rcParams[\"figure.figsize\"] = (9,7)\n",
+    "    for i in range(15):\n",
+    "        # define subplot\n",
+    "        ax = plt.subplot(3,5,i+1)\n",
+    "        ax.set_title(txt_labels[int(y[i])])\n",
+    "        # plot raw pixel data\n",
+    "        ax.imshow(X[i])\n",
+    "    # show the figure\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9dddf313",
+   "metadata": {},
+   "source": [
+    "Observe how there are only animals left in the training set `trainX` below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "486acf25",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_images(trainX,trainy)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7cf11eb8",
+   "metadata": {},
+   "source": [
+    "The test set on the other hand still visibily contains the non-animal images: `[ship, airplane, automobile, truck]`. If we consider `trainX` to be the representation of data, then these non-animal images in test dataset `testX` become outliers."
```

Comment:
careful about using term "representation", I'd only reserve that term as a synonym for the feature embeddings of the data

Reply:
In this case, I'd do: "the representation of data"
->
"representative of the normal data distribution"